packages/react-devtools-shared/src/events.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+export default class EventEmitter<Events: Object> {
+  listenersMap: Map<string, Set<Function>> = new Map();
+
+  addListener<Event: $Keys<Events>>(
+    event: Event,
+    listener: (...$ElementType<Events, Event>) => any,
+  ): void {
+    let listeners = this.listenersMap.get(event);
+    if (listeners === undefined) {
+      listeners = new Set();
+
+      this.listenersMap.set(event, listeners);
+    }
+
+    listeners.add(listener);
+  }
+
+  emit<Event: $Keys<Events>>(
+    event: Event,
+    ...args: $ElementType<Events, Event>
+  ): void {
+    const listeners = this.listenersMap.get(event);
+    if (listeners !== undefined) {
+      let caughtError = null;
```

Comment:
Nit: this won't work with `throw null`. While unlikely, it is more solid to have a separate `didThrow` boolean, and use that for the rethrow condition.

Reply:
yeah fair enough 😄